stan/math/prim/scal/meta/complex_step.hpp

```diff
@@ -0,0 +1,26 @@
+#ifndef STAN_MATH_PRIM_SCAL_META_COMPLEX_STEP_HPP
+#define STAN_MATH_PRIM_SCAL_META_COMPLEX_STEP_HPP
+
+#include <complex>
+#include <limits>
+#include <cmath>
+
+namespace stan {
+namespace math {
+
+template <class F> 
+// Complex step differentiation as described at: 
+// https://blogs.mathworks.com/cleve/2013/10/14/complex-step-differentiation/
+double complex_step(F f, double x) {
```

Comment:
Boost already has a function for this
https://www.boost.org/doc/libs/1_69_0/libs/math/doc/html/math_toolkit/diff.html